### PR TITLE
ENH: stats: Implement _sf and _isf for norminvgauss.

### LINKED ISSUE #8718: scipy.stats.norminvgauss.isf fails to converge for small probabilities
The inverse survivor function for `norminvgauss` fails to return a value for small probabilities (approx `p <= 1e-6` in the example below), instead generating a `RuntimeError`.

### Reproducing code example:
```
import scipy.stats
dst = scipy.stats.norminvgauss(1, 0)         # a=1, b=0
print(dst.sf(np.arange(0, 20, 2)))           # The survivor function can generate small probabilities
print(dst.isf(dst.sf(np.arange(0, 20, 2))))  # The isf can't always invert these small probabilities
```

This snippet prints the survivor probabilities but fails inverting them
```
[5.00000000e-01 2.72228574e-02 1.75561265e-03 1.43939011e-04
 1.33786541e-05 1.34098475e-06 1.41336775e-07 1.54399774e-08
 1.73257853e-09 1.98516092e-10]
```

### Error message:
```
RuntimeError                              Traceback (most recent call last)
<ipython-input-30-6c6d96d283f8> in <module>()
      2 dst = scipy.stats.norminvgauss(1, 0)
      3 print(dst.sf(np.arange(0, 20, 2)))
----> 4 print(dst.isf(dst.sf(np.arange(0, 20, 2))))

~/src/python/scipy/scipy/stats/_distn_infrastructure.py in isf(self, q)
    463 
    464     def isf(self, q):
--> 465         return self.dist.isf(q, *self.args, **self.kwds)
    466 
    467     def rvs(self, size=None, random_state=None):

~/src/python/scipy/scipy/stats/_distn_infrastructure.py in isf(self, q, *args, **kwds)
   1961             goodargs = argsreduce(cond, *((q,)+args+(scale, loc)))
   1962             scale, loc, goodargs = goodargs[-2], goodargs[-1], goodargs[:-2]
-> 1963             place(output, cond, self._isf(*goodargs) * scale + loc)
   1964         if output.ndim == 0:
   1965             return output[()]

~/src/python/scipy/scipy/stats/_distn_infrastructure.py in _isf(self, q, *args)
    903 
    904     def _isf(self, q, *args):
--> 905         return self._ppf(1.0-q, *args)  # use correct _ppf for subclasses
    906 
    907     # These are actually called, and should not be overwritten if you

~/src/python/scipy/scipy/stats/_distn_infrastructure.py in _ppf(self, q, *args)
    900 
    901     def _ppf(self, q, *args):
--> 902         return self._ppfvec(q, *args)
    903 
    904     def _isf(self, q, *args):

~/anaconda/envs/py36scipydev/lib/python3.6/site-packages/numpy/lib/function_base.py in __call__(self, *args, **kwargs)
   2753             vargs.extend([kwargs[_n] for _n in names])
   2754 
-> 2755         return self._vectorize_call(func=func, args=vargs)
   2756 
   2757     def _get_ufunc_and_otypes(self, func, args):

~/anaconda/envs/py36scipydev/lib/python3.6/site-packages/numpy/lib/function_base.py in _vectorize_call(self, func, args)
   2829                       for a in args]
   2830 
-> 2831             outputs = ufunc(*inputs)
   2832 
   2833             if ufunc.nout == 1:

~/src/python/scipy/scipy/stats/_distn_infrastructure.py in _ppf_single(self, q, *args)
   1591 
   1592         return optimize.brentq(self._ppf_to_solve,
-> 1593                                left, right, args=(q,)+args, xtol=self.xtol)
   1594 
   1595     # moment from definition

~/src/python/scipy/scipy/optimize/zeros.py in brentq(f, a, b, args, xtol, rtol, maxiter, full_output, disp)
    508     if rtol < _rtol:
    509         raise ValueError("rtol too small (%g < %g)" % (rtol, _rtol))
--> 510     r = _zeros._brentq(f,a,b,xtol,rtol,maxiter,args,full_output,disp)
    511     return results_c(full_output, r)
    512 

RuntimeError: Failed to converge after 100 iterations.
```

### Scipy/Numpy/Python version information:
```
1.1.0.dev0+512ac81 1.14.2 sys.version_info(major=3, minor=6, micro=3, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-8718.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
Here's what the example given by @pvanmulbregt in gh-8718 looks like with this PR:
```
In [11]: from scipy.stats import norminvgauss

In [12]: dst = norminvgauss(1, 0)

In [13]: k = np.arange(0, 20, 2)

In [14]: p = dst.sf(k)

In [15]: p
Out[15]: 
array([5.00000000e-01, 2.72228574e-02, 1.75561265e-03, 1.43939011e-04,
       1.33786541e-05, 1.34097487e-06, 1.41335621e-07, 1.54398419e-08,
       1.73256291e-09, 1.98514080e-10])

In [16]: dst.isf(p)
Out[16]: array([ 0.,  2.,  4.,  6.,  8., 10., 12., 14., 16., 18.])
```

**Comment by WarrenWeckesser** [comment]
The test failure on TravisCI is unrelated.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -3867,6 +3867,59 @@ def _pdf(self, x, a, b):
         sq = np.hypot(1, x)  # reduce overflows
         return fac1 * sc.k1e(a * sq) * np.exp(b*x - a*sq) / sq
 
+    def _sf(self, x, a, b):
+        if np.isscalar(x):
+            # If x is a scalar, then so are a and b.
```
Comment (chrisb83) [inline scipy/stats/_continuous_distns.py:3872]: why is that the case? does broadcasting happen before _sf is called?

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -395,6 +395,49 @@ def test_pdf_R(self):
         vals_pdf = stats.norminvgauss.pdf(x_test, a=1, b=0.5)
         assert_allclose(vals_pdf, r_pdf, atol=1e-9)
 
+    @pytest.mark.parametrize('x, a, b, sf, rtol',
+                             [(-1, 1, 0, 0.8759652211005315, 1e-13),
+                              (25, 1, 0, 1.1318690184042579e-13, 1e-4),
+                              (1, 5, -1.5, 0.002066711134653577, 1e-12),
+                              (10, 5, -1.5, 2.308435233930669e-29, 1e-9)])
+    def test_sf_isf_mpmath(self, x, a, b, sf, rtol):
```
Comment (chrisb83) [inline scipy/stats/tests/test_distributions.py:403]: maybe you can add a test to check that the _sf / _isf work for non-scalar input? you could reuse the same test cases, e.g.

sf([-1, 25], [1, 0], [0, 0])

**Review by chrisb83 — COMMENTED** [review]
just a few minor comments...

File: scipy/stats/_continuous_distns.py (hunk 1 above)
Comment (pvanmulbregt) [reply scipy/stats/_continuous_distns.py:3872]: The `argsreduce` call inside `rv_continuous.sf()` does the broadcasting before the call to `_sf()` is made.

**Review by mdhaber — COMMENTED** [review]
From gh-8718, 
>but I ended up not trying to come up with such a general solution.

why did you decide not implement this in general?

The general solution would address gh-2509.

**Comment by mdhaber** [comment]
The general solution would address gh-2509.

File: scipy/stats/tests/test_distributions.py (hunk 2 above)
Comment (mdhaber) [reply scipy/stats/tests/test_distributions.py:626]: @chrisb83 added the requested test. I think there has been enough time for comments, so if this still looks good to you, feel free to merge!

**Comment by chrisb83** [comment]
Hi Matt,

great, the changes look good. we can merge it once the two lint failures have been resolved:
```
scipy/stats/tests/test_distributions.py:668:15: E201 whitespace after '['
scipy/stats/tests/test_distributions.py:668:59: E261 at least two spaces before inline comment
```

File: scipy/stats/tests/test_distributions.py [hunk 3]
```diff
@@ -618,6 +618,68 @@ def test_pdf_R(self):
...
+        s = stats.norminvgauss.sf(x, a, b)
+        assert_allclose(s, sf, rtol=rtol)
+        i = stats.norminvgauss.isf(sf, a, b)
+        assert_allclose(i, x, rtol=rtol)
+
+    def test_sf_isf_mpmath_vectorized(self):
+        x = [-1, 25]
+        a = [1, 1]
+        b = 0
+        sf = [ 0.8759652211005315, 1.1318690184042579e-13] # see previous test
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:668]:
```suggestion
        sf = [0.8759652211005315, 1.1318690184042579e-13]  # see previous test
```

**Review by chrisb83 — APPROVED** [review]
The changes improve the accuracy of isf and sf. All tests pass.

**Pull request merged**